Makefile

```diff
@@ -1,3 +1,22 @@
+# Operating System detection and conditional compile options
+
+ifeq ($(OS),Windows_NT)
+    OSYSTEM := Windows
+else
+    OSYSTEM := $(shell sh -c 'uname 2>/dev/null || echo Unknown')
+endif
+
+ifeq ($(OSYSTEM),Windows) # MS Windows
+    GCOPT +=
+endif
+
+ifeq ($(OSYSTEM),Darwin) # Apple macOS
+    GCOPT += -I/opt/homebrew/include -L/opt/homebrew/lib -Wno-unknown-pragmas
```

Comment:
Ah! I didn't mean to include the `-Wno-unknown-pragmas` switch as it was an experiment to see why m65fdisk won't build. I'll make another commit for m65fdisk fixes if I can work out why it won't build but I notice it isn't built when this is a submodule for mega65-core so perhaps it is not actually needed.